controller/workitem.go

```diff
@@ -465,21 +466,26 @@ func ConvertJSONAPIToWorkItem(appl application.Application, source app.WorkItem,
 			target.Fields[workitem.SystemIteration] = iterationUUID.String()
 		}
 	}
-	if source.Relationships != nil && source.Relationships.Area != nil {
-		if source.Relationships.Area.Data == nil {
-			delete(target.Fields, workitem.SystemArea)
-		} else {
-			d := source.Relationships.Area.Data
-			areaUUID, err := uuid.FromString(*d.ID)
-			if err != nil {
-				return errors.NewBadParameterError("data.relationships.area.data.id", *d.ID)
-			}
-			if _, err = appl.Areas().Load(context.Background(), areaUUID); err != nil {
-				return errors.NewBadParameterError("data.relationships.area.data.id", *d.ID)
-			}
-			target.Fields[workitem.SystemArea] = areaUUID.String()
+
+	if source.Relationships != nil && source.Relationships.Area != nil && source.Relationships.Area.Data != nil {
+		d := source.Relationships.Area.Data
+		areaUUID, err := uuid.FromString(*d.ID)
+		if err != nil {
+			return errors.NewBadParameterError("data.relationships.area.data.id", *d.ID)
+		}
+		if _, err = appl.Areas().Load(context.Background(), areaUUID); err != nil {
+			return errors.NewBadParameterError("data.relationships.area.data.id", *d.ID)
 		}
+		target.Fields[workitem.SystemArea] = areaUUID.String()
+	} else if appl != nil && (source.Relationships == nil || source.Relationships.Area == nil || source.Relationships.Area.Data == nil) {
```

Comment:
hum, are you saying that if `source.Relationships == nil` then this means that the client did not request any change in the work item relation ships at all ? In that case, yes, I'd rather follow your suggestion.